### PR TITLE
Normalization

### PR DESCRIPTION
**Pull request by befelix**
Currently the normalization is only applied if one has more than 1 data point. This can lead to a strange corner cases (in the test) where the optimization fails. This PR makes sure that the mean is always sutracted, even when `std=0`. This can help with miss-specified models. Also specifying a wrong normalization type now raises a more useful error.

Granted, the problem this PR fixes will not come up in practice often, but it's still useful for plotting illustrative examples. 

Also this PR fixes plotting of the acquisition function when the model is not initialized. Previously normalization was not applied there.

### CODE REVIEW

File: GPyOpt/core/bo.py [hunk 1]
```diff
@@ -228,29 +228,47 @@ def _compute_next_evaluations(self, pending_zipped_X=None, ignored_zipped_X=None
...
+            Y_norm = Y - Y.mean()
+            std = Y.std()
+            if std > 0:
+                Y_norm /= std
+        elif normalization_type == 'maxmin':
+            Y_norm = Y - Y.min()
+            y_range = np.ptp(Y)
+            if y_range > 0:
+                Y_norm /= y_range
+                # [0, 1] is a strange range for a zero-mean GP
```
Comment (apaleyes) [inline GPyOpt/core/bo.py]: It might be strange indeed, but it is now a behavior of the library. I would be very cautions changing it here. Let's separate things out: this PR is dealing with applying normalization to 1 point set, and let it be so. Changing the normalization range is a separate change, so if you still believe it is the right thing to do, can you please post a separate PR with just that so we can have a focused discussion?

File: GPyOpt/core/bo.py [hunk 2]
```diff
@@ -269,7 +287,11 @@ def plot_acquisition(self,filename=None):
         if self.model.model is None:
             from copy import deepcopy
             model_to_plot = deepcopy(self.model)
-            model_to_plot.updateModel(self.X, self.Y, self.X, self.Y)
+            if self.normalize_Y:
```
Comment (apaleyes) [inline GPyOpt/core/bo.py]: Good catch

**Comment by befelix** [comment]
I've changed the things you've requested - should be good to merge now.

I'm writing a paper about hyperparameter optimization for BO and decided to base the method on `GPyOpt` to make it easy to compare against MAP & MCMC. These issues mostly popped up when reading through the code to make sure there's nothing strange going on in the library :)

File: GPyOpt/testing/util_tests/test_general_utils.py [hunk 3]
```diff
@@ -117,3 +121,51 @@ def test_sobol_design(self):
         samples = initial_design('sobol', self.design_space, init_points_count)
         self.assertEqual(len(samples), init_points_count)
         self.assert_samples_against_space(samples)
+
+
+class TestNormalization(unittest.TestCase):
```
Comment (apaleyes) [inline GPyOpt/testing/util_tests/test_general_utils.py]: That is very nice set of tests, thanks!
